src/sage/rings/ideal.py

```diff
@@ -749,17 +754,17 @@ def is_primary(self, P=None):
 
         Some examples from the Macaulay2 documentation::
 
-            sage: R.<x, y, z> = GF(101)[]
-            sage: I = R.ideal([y^6])
-            sage: I.is_primary()
+            sage: R.<x, y, z> = GF(101)[]                                               # optional - sage.rings.finite_rings
+            sage: I = R.ideal([y^6])                                                    # optional - sage.rings.finite_rings
+            sage: I.is_primary()                                                        # optional - sage.rings.finite_rings
             True
```

Comment:
`sage.rings.finite_rings` depends on singular?

Reply:
It doesn't; so I'll add the `# optional - sage.libs.singular`.